src/Lib/Buddy.php

```diff
@@ -0,0 +1,245 @@
+			} elseif ($this->origMsg !== '') {
+				$message = $this->origMsg;
+			}
+		}
+
+		return ['type' => 'http response', 'message' => $message, 'error' => $error];
+	}
+
+	/**
+	 * Checking if request can be handled by Buddy
+	 *
+	 * @return array<ACTION>
+	 */
+	protected function getHandleActions(): array {
+		$errorType = $this->detectErrorType();
+		if (isset(static::ACTION_MAP[$errorType])) {
+			$queryType = $this->detectQueryType();
+			return static::ACTION_MAP[$errorType][$queryType] ?? [];
+		}
+		return [];
+	}
+
+	/**
+	 * @return string
+	 */
+	protected function detectErrorType(): string {
+		// so far only use case with non-existing local index
+		if (preg_match('/index (.*?) absent/', $this->origMsg)) {
+			return 'NO_INDEX';
+		}
+		if (strpos($this->origMsg, 'unexpected identifier') !== false) {
+			return 'UNKNOWN_COMMAND';
+		}
+		return '';
+	}
+
+	/**
+	 * @return string
+	 */
+	protected function detectQueryType(): string {
+		if (stripos($this->query, 'INSERT') === 0) {
+			return 'INSERT_QUERY';
+		}
+		if (stripos($this->query, 'SHOW QUERIES') === 0) {
+			return 'SHOW_QUERIES_QUERY';
+		}
+		return '';
+	}
+
+	/**
+	 * @param ACTION $action
+	 * @return string
+	 */
+	protected function buildMntRequest(ACTION $action): string {
+		switch ($action) {
+			case ACTION::INSERT:
+				return $this->query;
+			case ACTION::CREATE_INDEX:
+				return $this->buildMntRequestWithHelpers('INSERT', 'CREATE');
+			case ACTION::SELECT_SYSTEM_SESSIONS:
+				return 'SELECT connid AS ID, `last cmd` AS query, host, proto FROM @@system.sessions)';
+			default:
+				return '';
+		}
+	}
+
+	/**
+	 * @param string $parseStmt
+	 * @param string $buildStmt
+	 * @return string
+	 * @throws Throwable
+	 */
+	protected function buildMntRequestWithHelpers(string $parseStmt, string $buildStmt): string {
+		try {
+			if (!method_exists($this->queryParserLocator, 'getQueryParser')) {
+				$this->error('Required parser cannot be found');
+			} else {
+				$parser = $this->queryParserLocator->getQueryParser($this->reqFormat, $parseStmt);
+				if (method_exists($parser, 'parse')) {
+					$queryData = $parser->parse($this->query);
+					if (array_key_exists('error', $queryData)) {
+						return $queryData['error'];
+					}
+					if (!method_exists($this->stmtBuilder, 'build')) {
+						$this->error('Statement builder cannot be found');
+					}
+					return $this->stmtBuilder->build($buildStmt, $queryData);
+				}
+				$this->error('Unvalid parser detected');
+			}
+		} catch (Throwable $e) {
+			$this->error($e->getMessage());
+		}
+		return '';
+	}
+
+	/**
+	 * @param string $mntRequest
+	 * @return string
+	 */
+	protected function getMntResponse(string $mntRequest): string {
+		$resp  = '';
+		$respEndpoint = "http://{$this->clAddrPort}/cli";
+		$conn = curl_init($respEndpoint);
+		if ($conn === false) {
+			$this->error("Cannot conect to server at $respEndpoint");
+		} else {
+			curl_setopt($conn, CURLOPT_RETURNTRANSFER, true);
```

Comment:
Prefer to use `curl_setopt_array` for a single call to set all options from an array — much clearer.